backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceEventsHandler.java

```diff
@@ -0,0 +1,202 @@
+package backtraceio.library.metrics;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Message;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import backtraceio.library.common.BacktraceMathHelper;
+import backtraceio.library.common.BacktraceSerializeHelper;
+import backtraceio.library.interfaces.Api;
+import backtraceio.library.logger.BacktraceLogger;
+import backtraceio.library.models.BacktraceMetricsSettings;
+import backtraceio.library.services.BacktraceHandlerThread;
+import backtraceio.library.services.BacktraceReportSender;
+
+public abstract class BacktraceEventsHandler extends Handler {
```

Comment:
Wouldn't it be better if we will use <T> parameter where T is type event? With generic type we will avoid using reflection to be sure we store in the event list T parameters.

Reply:
I tried that earlier but it got a bit messy. I will take another look as I refactor some of this stuff

Reply:
Feel free to ask me anything! I love generics <3

Reply:
I might ask you some more questions about this offline as I do this refactor, thank you @konraddysput !